common/state/state_kv.cpp

```diff
@@ -172,6 +239,100 @@ pstate::block_offset_t* pstate::trie_node::goto_next_offset(trie_node_header_t*
     return NULL;
 }
 
+bool pstate::free_space_collector::allocate(const unsigned int& length, block_offset_t& out_bo)
+{
+    bool space_found = false;
+
+    SAFE_LOG(PDO_LOG_DEBUG, "search for space to be reused");
+    for(auto it = free_space_collection.begin(); it != free_space_collection.end(); it++)
+    {
+        if(it->length >= length)
+        {
+            space_found = true;
+            SAFE_LOG(PDO_LOG_DEBUG, "found: blocknum %u bytes %d len %u", it->bo.block_num, it->bo.bytes, it->length);
+            //return the block offset
+            out_bo = it->bo;
+
+            if(it->length == length)
+            {
+                //the requested length match, so remove item
+                free_space_collection.erase(it);
+                SAFE_LOG(PDO_LOG_DEBUG, "space allocated, item removed");
+            }
```

Comment:
please only leave the most necessary, high level debugging logging in the code. you can cherry pick updates to avoid commiting the logging if necessary.

Reply:
done